### PR TITLE
Switch alarm_log to use the buffer json functions

### PR DESCRIPTION
**Pull request by stelfrag**
##### Summary
Switch the alarm_log api endpoint to use the new buffer json functions.
The `/api/v1/alarm_log` endpoint output should be the same before and after this PR

### CODE REVIEW

File: database/sqlite/sqlite_health.c [hunk 1]
```diff
@@ -1303,7 +1301,7 @@ void sql_health_alarm_log2json(RRDHOST *host, BUFFER *wb, time_t after, const ch
 
          if (unlikely(rc != SQLITE_OK)) {
             error_report("Failed to prepare statement SQL_SELECT_HEALTH_LOG");
-            buffer_strcat(wb, "\n]");
+//            buffer_strcat(wb, "\n]");
```
Comment (thiagoftsm) [inline database/sqlite/sqlite_health.c:1304]: Hello @stelfrag ,

why are you commenting instead to remove?

Best regards!

File: database/sqlite/sqlite_health.c [hunk 2]
```diff
@@ -1352,129 +1353,73 @@ void sql_health_alarm_log2json(RRDHOST *host, BUFFER *wb, time_t after, const ch
...
+                  buffer_json_member_add_int64(wb, "unique_id",            (int64_t) sqlite3_column_int64(stmt_query, 0));
+                  buffer_json_member_add_int64(wb, "alarm_id",             (int64_t) sqlite3_column_int64(stmt_query, 1));
+                  buffer_json_member_add_int64(wb, "alarm_event_id",       (int64_t) sqlite3_column_int64(stmt_query, 2));
+        buffer_json_member_add_string_or_empty(wb, "config_hash_id",       config_hash_id);
+        buffer_json_member_add_string_or_empty(wb, "transition_id",        transition_id);
+        buffer_json_member_add_string_or_empty(wb, "name",                 (const char *) sqlite3_column_text(stmt_query, 12));
+        buffer_json_member_add_string_or_empty(wb, "chart",                (const char *) sqlite3_column_text(stmt_query, 13));
+        buffer_json_member_add_string_or_empty(wb, "context",              (const char *) sqlite3_column_text(stmt_query, 29));
+        buffer_json_member_add_string_or_empty(wb, "class",                sqlite3_column_text(stmt_query, 26) ? (const char *) sqlite3_column_text(stmt_query, 26) : (char *) "Unknown");
```
Comment (thiagoftsm) [inline database/sqlite/sqlite_health.c:1369]: You are using `Unknown` different times, probably it is better to set it as a macro helping old compilers to optimize the code.

File: database/sqlite/sqlite_health.c [hunk 3]
```diff
@@ -1352,129 +1353,73 @@ void sql_health_alarm_log2json(RRDHOST *host, BUFFER *wb, time_t after, const ch
...
+        buffer_json_member_add_string_or_empty(wb, "old_value_string",     sqlite3_column_type(stmt_query, 24) == SQLITE_NULL ? "-" :
+                                                                                      format_value_and_unit(old_value_string, 100, sqlite3_column_double(stmt_query, 24), (char *) sqlite3_column_text(stmt_query, 17), -1));
+                  buffer_json_member_add_int64(wb, "last_repeat",          (int64_t)sqlite3_column_int64(stmt_query, 25));
+                buffer_json_member_add_boolean(wb, "silenced",             (sqlite3_column_int64(stmt_query, 9) & HEALTH_ENTRY_FLAG_SILENCED));
+        buffer_json_member_add_string_or_empty(wb, "summary",              (const char *) sqlite3_column_text(stmt_query, 31));
+        buffer_json_member_add_string_or_empty(wb, "info",                 (const char *) sqlite3_column_text(stmt_query, 18));
+                buffer_json_member_add_boolean(wb, "no_clear_notification",(sqlite3_column_int64(stmt_query, 9) & HEALTH_ENTRY_FLAG_NO_CLEAR_NOTIFICATION));
+
```
Comment (thiagoftsm) [inline database/sqlite/sqlite_health.c:1400]: Is there a special reason to use `if ... else` instead ternary operator for next two ?

**Comment by thiagoftsm** [comment]
@stelfrag, please, take a look in the error reported by the unit test. I do not remember to see it with other PR https://github.com/netdata/netdata/pull/16321.

The `alarm_log` output is correct, so I am waiting for answers before to proceed.

File: database/sqlite/sqlite_health.c (hunk 1 above)
Comment (stelfrag) [reply database/sqlite/sqlite_health.c:1304]: i will remove it

File: database/sqlite/sqlite_health.c (hunk 3 above)
Comment (stelfrag) [reply database/sqlite/sqlite_health.c:1400]: it is using `buffer_json_member_add_string` and `buffer_json_member_add_double`

**Review by thiagoftsm — APPROVED** [review]
PR is working as expected, LGTM!

**Review by MrZammler — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
PR is working as expected, LGTM!

**Pull request merged**